src/core/thread/dua_manager.cpp

```diff
@@ -33,51 +33,106 @@
 
 #include "dua_manager.hpp"
 
-#if (OPENTHREAD_FTD || OPENTHREAD_MTD) && OPENTHREAD_CONFIG_DUA_ENABLE
+#if OPENTHREAD_CONFIG_DUA_ENABLE || OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
 
 #include "common/code_utils.hpp"
 #include "common/instance.hpp"
 #include "common/locator-getters.hpp"
 #include "common/logging.hpp"
 #include "common/settings.hpp"
 #include "net/ip6_address.hpp"
+#include "thread/mle_types.hpp"
 #include "thread/thread_netif.hpp"
+#include "thread/thread_tlvs.hpp"
+#include "thread/thread_uri_paths.hpp"
 #include "utils/slaac_address.hpp"
 
 namespace ot {
 
 DuaManager::DuaManager(Instance &aInstance)
     : InstanceLocator(aInstance)
+    , Notifier::Receiver(aInstance, DuaManager::HandleNotifierEvents)
+    , mTimer(aInstance, DuaManager::HandleTimer, this)
+    , mRegistrationTask(aInstance, DuaManager::HandleRegistrationTask, this)
+    , mDuaNotification(OT_URI_PATH_DUA_REGISTRATION_NOTIFY, &DuaManager::HandleDuaNotification, this)
+    , mIsDuaPending(false)
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    , mDuaState(DuaState::kNotExist)
     , mDadCounter(0)
+    , mLastRegistrationTime(0)
+#endif
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    , mChildIndexDuaRegistering(0)
+    , mRegisterCurrentChildIndex(false)
```

Comment:
```suggestion
    , mChildIndexDuaRegistering(0)
    , mRegisterCurrentChildIndex(false)
```

Since the value of `mChildIndexDuaRegistering` is only valid when `mRegisterCurrentChildIndex=true`, I think we can combine them into one variable: `mChildIndexDuaRegistering`. We can set it to `MaxChildIndex+1` as an invalid value.

Reply:
It is an option. But I still think a separate variable be more friendly for reader. To avoid extra consumption on RAM, will update to make `mChildIndexDuaRegistering` and  `mRegisterCurrentChildIndex` share the same 2 bytes.

Reply:
Fine to keep them separate.